pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -361,8 +362,10 @@ private static Expression updateComparisonPredicate(Expression expression) {
         default:
           List<Expression> operands = functionCall.getOperands();
           List<Expression> newOperands = new ArrayList<>();
-          for (int i = 0; i < operands.size(); i++) {
-            newOperands.add(updateComparisonPredicate(operands.get(i)));
+          if (operands != null) {
+            for (int i = 0; i < operands.size(); i++) {
```

Comment:
Not related to this PR -- JVM won't inline this. Let's just do `int size = operands.size()` before the loop and `i < size`

Reply:
operands is null so operands.size will also throw exception, used operandsSize() instead